smac/facade/bogp_facade.py

```diff
@@ -61,41 +66,46 @@ def __init__(self, model_type='gp_mcmc', **kwargs):
 
             if model_type == "gp":
-                model = GaussianProcess(
-                    types=types,
-                    bounds=bounds,
-                    kernel=kernel,
-                    prior=prior,
-                    rng=rng,
-                    normalize_output=True,
-                    normalize_input=True,
-                )
+                model_class = GaussianProcess
+                kwargs['model'] = model_class
+                model_kwargs = kwargs.get('model_kwargs', dict())
+                model_kwargs['kernel'] = kernel
+                model_kwargs['prior'] = prior
+                model_kwargs['normalize_input'] = True
+                model_kwargs['normalize_output'] = True
+                model_kwargs['normalize_input'] = True
+                model_kwargs['seed'] = rng.randint(0, 2 ** 20)
             elif model_type == "gp_mcmc":
-                model = GaussianProcessMCMC(
-                    types=types,
-                    bounds=bounds,
-                    kernel=kernel,
-                    prior=prior,
-                    n_hypers=n_hypers,
-                    chain_length=200,
-                    burnin_steps=100,
-                    normalize_input=True,
-                    normalize_output=True,
-                    rng=rng,
-                )
-            kwargs['model'] = model
+                model_class = GaussianProcessMCMC
+                kwargs['model'] = model_class
+                model_kwargs = kwargs.get('model_kwargs', dict())
+                model_kwargs['kernel'] = kernel
+                model_kwargs['prior'] = prior
+                model_kwargs['n_hypers'] = n_hypers
+                model_kwargs['chain_length'] = 200
+                model_kwargs['burnin_steps'] = 100
+                model_kwargs['normalize_input'] = True
+                model_kwargs['normalize_output'] = True
+                model_kwargs['seed'] = rng.randint(0, 2**20)
+            kwargs['model_kwargs'] = model_kwargs
+
         super().__init__(**kwargs)
 
         if self.solver.scenario.n_features > 0:
             raise NotImplementedError("BOGP cannot handle instances")
 
         self.logger.info(self.__class__)
 
-        self.solver.random_configuration_chooser.prob = 0.0
+        # assumes random chooser for random configs
+        random_config_chooser_kwargs = kwargs.get('random_configuration_chooser_kwargs', dict())
+        random_config_chooser_kwargs['prob'] = random_config_chooser_kwargs.get('prob', 0.0)
+        kwargs['random_configuration_chooser_kwargs'] = random_config_chooser_kwargs
 
         # only 1 configuration per SMBO iteration
-        self.solver.scenario.intensification_percentage = 1e-10
-        self.solver.intensifier.min_chall = 1
+        intensifier_kwargs = kwargs.get('intensifier_kwargs', dict())
+        intensifier_kwargs['min_chall'] = 1
+        kwargs['intensifier_kwargs'] = intensifier_kwargs
+        scenario.intensification_percentage = 1e-10
```

Comment:
I'm really wondering how we can avoid such hacking of the scenario object. Maybe we can also alter it earlier? Or make all such "scenario" arguments which change the behavior of SMAC (and therefore aren't scenario arguments) be passed via the kwargs, and then handled by the respective facades? Then, only the "real" scenario arguments are handed around in the SMAC code base.